app/models/instrument_processes_instrument.rb

```diff
@@ -28,7 +28,7 @@ def self.find_partial_name!(instrument_barcode, instrument_process_id)
 
   def self.find_from_instrument_barcode_and_instrument_process_id(instrument_barcode, instrument_process_id)
     instrument = Instrument.find_by_barcode(instrument_barcode)
-    process = InstrumentProcess.find(instrument_process_id)
+    process = InstrumentProcess.find_by_id(instrument_process_id)
```

Comment:
Don't remember this change - what was that about?

Reply:
This was me tidying up the initial screen behaviour. If you pressed submit without selecting all the options you got an exception because it tried to look up a nil instrument process id using find. find_by_id doesn't throw the exception, the code falls into the intended error handling that was already there and you get the intended result (screen does nothing). Presumably due to a change in the way find worked since the code was written.

Reply:
Ah ok, thanks!
Weird that one throws an exception and one doesn't.